File: C++ Programming/problems on file handling/problem7.cpp
/*
7. Write a program which accepts file name from user and count number of small characters from that file. 
	
	Input : Demo.txt 
	
	Output : Number of small characters are 21 

	Author : Milind Nalawade

	Date : 1/09/2021

	Language : C++ programming
	
*/

using namespace std;
#include<iostream>
#include<stdlib.h>
#include<unistd.h>
#include<fcntl.h>

//Entry point function
int main()
{
	int fd=0,ret=0,cnt=0,i=0;
	char arr[100]={'\0'};
	
	cout<<"\n Enter the file name : ";
	cin.getline(arr,100);

	fd=open(arr,O_RDONLY|O_CREAT,0777);
	if(fd==-1)
	{
		cout<<"\n file unalbe to open\n";
	close(fd);
	}

	while((ret=read(fd,arr,100))!=0)
	{
		while(arr[i]!='\0')
		{
			if(arr[i]>=97 && arr[i]<=122)
			{
				cnt++;
			}
			i++;
		}			
	}
	cout<<"\n Number of small characters are : "<<cnt;
	return 0;

}

File: Java Programming/problems on bitwise operator/problem7.java
/*

7. Write a program which accept one number from user and off 7th and 10th bit of that number.
	Return modified number. 
	
	Input :  577
	Output : 1 
	
	Author : Milind Nalawade
	
	Date : 02/08/2021
	
	Language : Java 14.0.2 

*/

import java.lang.*;
import java.util.*;

class Logic
{
	public static int ChkBits(int ino)
	{
		int pos=0x00000240;
		int flag=pos&ino;
		if(flag==pos)
		{
			return ino^pos;
		}
		return ino;
	}
}

class problem7 extends Logic
{
	public static void main(String args[])
	{
		Scanner sobj=new Scanner (System.in);
		System.out.print("\n Enter your number : ");
		int no=sobj.nextInt();
		int ret=ChkBits(no);
		System.out.print("\n Output : "+ret+"\n");
	}
}